core/templates/base-components/oppia-root.directive.ts

```diff
@@ -44,8 +44,10 @@ angular.module('oppia').directive('oppiaRoot', ['$translate',
             );
             i18nLanguageCodeService.setI18nLanguageCode(
               $translate.proposedLanguage() || $translate.use());
+
             // The next line allows the transcluded content to start executing.
             $scope.initialized = true;
+            $scope.$apply();
```

Comment:
@seanlip do you have any thoughts on why this $apply might be needed?

Reply:
$apply is needed now because of the promise that has been introduced in this pr in oppia-angular-root (line no. 1723). It is not a problem in angular because angular uses zone.js, that monkey patches these promises to automatically trigger change detection when they resolve. However, angularjs doesn't use zonejs so we have to manually trigger change detection by calling $scope.$apply(). Hope this clarifies why we need this $apply now.

Reply:
Aha, thanks for the explanation @srijanreddy98!

@MegrezZhu suggest adding a brief comment capturing the high-level aspects of that here to make sure the $apply stays in place.